### PR TITLE
fix: Allow identities in PauliString observable

### PR DESCRIPTION
**Pull request by speller26**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit_tests/braket/quantum_information/test_pauli_string.py [hunk 1]
```diff
@@ -34,15 +34,15 @@
...
+    "pauli_string, string, phase, observable, obs_with_id",
     [
-        ("+XZ", "+XZ", 1, X() @ Z()),
-        ("-ZXY", "-ZXY", -1, Z() @ X() @ Y()),
-        ("YIX", "+YIX", 1, Y() @ X()),
-        (PauliString("-ZYXI"), "-ZYXI", -1, Z() @ Y() @ X()),
+        ("+XZ", "+XZ", 1, X() @ Z(), X() @ Z()),
+        ("-ZXY", "-ZXY", -1, Z() @ X() @ Y(), Z() @ X() @ Y()),
+        ("YIX", "+YIX", 1, Y() @ X(), Y() @ I() @ X()),
+        (PauliString("-ZYXI"), "-ZYXI", -1, Z() @ Y() @ X(), Z() @ Y() @ X() @ I()),
```
Comment (shpface) [inline test/unit_tests/braket/quantum_information/test_pauli_string.py:42]: Should we attempt to cover some more edge cases with these tests? such as repeated operators or only Identity operators?

**Review by shpface — APPROVED** [review]
